### PR TITLE
chore(flamegraphs): align naming and dir structures

### PR DESCRIPTION
**Pull request by JonasBa**
This was a bit all over the place, I reorganized it so it follows our structure better. Also renamed some stuff to follow the flamegraph* naming

### CODE REVIEW

File: static/app/components/profiling/flamegraph/flamegraph.tsx [hunk 1]
```diff
@@ -42,8 +41,9 @@ import {formatTo, ProfilingFormatterUnit} from 'sentry/utils/profiling/units/uni
 import {useDevicePixelRatio} from 'sentry/utils/useDevicePixelRatio';
 import {useMemoWithPrevious} from 'sentry/utils/useMemoWithPrevious';
 
-import {FlamegraphWarnings} from './FlamegraphWarnings';
-import {ProfilingFlamechartLayout} from './profilingFlamechartLayout';
+import {FlamegraphDrawer} from './flamegraphDrawer/flamegraphDrawer';
+import {FlamegraphWarnings} from './flamegraphOverlays/FlamegraphWarnings';
+import {FlamegraphLayout} from './flamegraphLayout';
```
Comment (Zylphrex) [inline static/app/components/profiling/flamegraph/flamegraph.tsx:46]: Any reason we're going back to the flamegraph terminology? This mix of flamegraph vs flamechart terminology can be confusing for people unfamiliar with it.

**Review by Zylphrex — APPROVED** [review]

File: static/app/components/profiling/flamegraph/flamegraph.tsx (hunk 1 above)
Comment (JonasBa) [reply static/app/components/profiling/flamegraph/flamegraph.tsx:46]: Yeah, lets stick to graph as it sounds like that is where the future direction might go. I also dont think there is much preventing us today from actually rendering a graph if we wanted to

File: static/app/components/profiling/flamegraph/flamegraph.tsx [hunk 2]
```diff
@@ -355,8 +355,8 @@ function Flamegraph(props: FlamegraphProps): ReactElement {
             />
           </ProfileDragDropImport>
         }
-        frameStack={
-          <FrameStack
+        flamechartDrawer={
```
Comment (Zylphrex) [inline static/app/components/profiling/flamegraph/flamegraph.tsx:358]: This is still calling it flamechart and the props above. Don't have a strong opinion in either direction, but if we can stick with 1 that'd be great.

Comment (JonasBa) [reply static/app/components/profiling/flamegraph/flamegraph.tsx:358]: Yup, let me rename that too

**Pull request merged**